mycore-pi/src/main/java/org/mycore/pi/urn/rest/MCRDerivateURNUtils.java

```diff
@@ -83,7 +83,7 @@ public static URL getURL(MCRPIRegistrationInfo piInfo) {
                     String filePath = "/" + file.getOwner() + "/" + file.getFileName();
                     return new URL(
                         MCRFrontendUtil.getBaseURL() + "servlets/" + MCRFileNodeServlet.class.getSimpleName()
-                            + filePath);
+                            + filePath.replaceAll(" ", "%20"));
```

Comment:
Do not URL encode like this. There will be more cases of characters that need to be percent encoded.